elastic/kibana
# [ Security Solution ]Re-organize the Investigations api integration test for MKI

## Summary

This is Part 1 of the in resolving the issue : https://github.com/elastic/kibana/issues/183645 .

This PR re-organizes investigations API tests so that they can be run in Serveless MKI at both `basic/essentials` and `complete` licenses.

## How to test this PR 

Below are the commands that are affected by this change and you can test the PR by running  below commands.

Each commands sets up the test environment and  give you a command to run tests. Please run those tests to see if everything is okay. An example is shown in below screenshot.

<img width="1916" alt="grafik" src="https://github.com/user-attachments/assets/fa400450-e4aa-41dc-a1ea-ac21634c46d3">


|Module|Deployment|License|Command|
|--|--|--|--|
|Timelines|ESS|basic|`yarn investigations:basic:timeline:server:ess`|
|Timelines|ESS|Trial|`yarn investigations:timeline:server:ess`|
|Timelines|Serverless|basic|`yarn investigations:basic:timeline:server:serverless`|
|Timelines|Serverless|Trial|`yarn investigations:timeline:server:serverless`|
|Saved Objects|ESS|basic|`yarn investigations:basic:saved-objects:server:ess`|
|Saved Objects|ESS|Trial|`yarn investigations:saved-objects:server:ess`|
|Saved Objects|Serverless|basic|`yarn investigations:basic:saved-objects:server:serverless`|
|Saved Objects|Serverless|Trial|`yarn investigations:saved-objects:server:serverless`|
